src/parser.cc

```diff
@@ -773,20 +773,28 @@ bool readFileInternal(const std::string &_filename, const bool _convert,
   {
     return true;
   }
-  else if (URDF2SDF::IsURDF(filename))
+  else
   {
-    URDF2SDF u2g;
-    auto doc = makeSdfDoc();
-    u2g.InitModelFile(filename, _config, &doc);
-    if (sdf::readDoc(&doc, _sdf, filename, _convert, _config, _errors))
+    tinyxml2::XMLElement *robotXml = xmlDoc.FirstChildElement("robot");
+    if (robotXml && URDF2SDF::IsURDF(filename))
```

Comment:
Calling `IsURDF` seems unnecessary at this point and wastes cycles since it actually fully parses the file. IMO, if the first tag is `robot`, we can assume it's a URDF and proceed.

Reply:
Sounds about right. I was not sure how much removing this parsing would modify behavior though so I opted to minimize changes, if you think it's OK I'll remove it then.

Reply:
Removed in https://github.com/gazebosim/sdformat/pull/1290/commits/029b204e98cf5c01399b8e4297bea4f1dbaeac20.